### ISSUE TITLE
authenticating user 

### ISSUE THREAD

**Issue by daksamit** [strapi/strapi#3106]
**Informations**
- **Node.js version**: v11.1.0
- **NPM version**: 6.4.1
- **Strapi version**: 3.0.0-alpha.25.2
- **Database**: mongo
- **Operating system**: linux 
- Strapi is running with docker-strapi

**What is the current behavior?**

Following error occurs when try to login as 'Authenticated' user (not 'Administrator' user role): 
`[ { "messages": [ { "id": "Auth.form.error.noAdminAccess" } ] } ]`

while trying to:
`POST http://localhost:1337/auth/local`

js fetch request:
```
fetch('http://localhost:1337/auth/local', {
        method: 'post',
        headers: {
          'Content-Type': 'application/json'
        },
        body: JSON.stringify({
          identifier: 'username',
          password: 'password'
        })
      })
```

Is it normal behavior?

I would like to get user and jwt.

The same thing happens while trying to authenticate with following guide:
https://strapi.io/documentation/3.x.x/guides/authentication.html#login

**Comment by derrickmehaffy** [comment]
can you try to console.log() that `body` the only thing that comes to mind is the structure of the body is wrong. Also can you test via postman

**Comment by daksamit** [comment]
It returns `400 (Bad Request)` for user with Authenticated role. it works for Administrator.
Tried also [strapi-sdk-javascript](https://github.com/strapi/strapi-sdk-javascript) with " `strapi.login(username, password)` "

**Comment by derrickmehaffy** [comment]
You may have a problem with your permissions, also what is the error in the strapi console?

**Comment by daksamit** [comment]
![image](https://user-images.githubusercontent.com/7165997/55796450-58219400-5aca-11e9-9f2d-69d27a75ccff.png)

Default role is Public, yes ?
In admin panel under Roles & Permissions -> USERS PERMISSIONS -> Auth:
callback points to /auth/local and when set to `none` - that thing happens.
When I tried set `isAuthenticated` (test case) I have `401 Unauthorized` instead.

**Comment by derrickmehaffy** [comment]
@daksamit you can't set `isAuthenticated` on any route in the public role. It is like trying to divide by zero. You can't be unauthenticated and require auth at the same time. :wink:

**Comment by daksamit** [comment]
@derrickmehaffy I know exactly. Just for test case.

strapi/plugins/users-permissions/controllers/Auth.js:68 points request:
```
      if (user.role.type !== 'root' && ctx.request.admin) {
        return ctx.badRequest(null, ctx.request.admin ? [{ messages: [{ id: 'Auth.form.error.noAdminAccess' }] }] : `You're not an administrator.`);
      }
```
I think it's because of ctx.request.admin ? 
I don't want to login as admin. Should I send more info in body ?

**Comment by derrickmehaffy** [comment]
@daksamit that line of code is for the AdminUI. Non-admin users can't login to the adminUI

**Comment by daksamit** [comment]
It makes sense, but there is out-of-the-box solution to authenticate user ?
I just want to get jwt token for user and store it on client side.

**Comment by derrickmehaffy** [comment]
That is what /auth/local is for, I'm guessing somewhere along the line something got changed that shouldn't have because I can't reproduce this at all.

**Comment by daksamit** [comment]
It's strange. I will try with fresh docker setup or without it maybe I'll find solution.

**Comment by daksamit** [comment]
OK, I solved the problem.

### TL;DR

CORS policy, custom headers, application/json, OPTIONS request

### Solution

What's really strange I don't understand, you can log in into admin user without problems I didn't realize that's because of CORS.
Moreover it is not working with strapi-sdk-javascript `strapi.login(username, password)` what confused me.

I didn't get in depth but I think it should be fixed.

**Comment by soupette** [comment]
@daksamit since your problem seems to be solved and it is not directly related with Strapi itself I'm closing the issue.

**Comment by daksamit** [comment]
Another solution: 
a requested to api from http://localhost:4000 even strapi hasn't been running on this port.
I'm aware this port is also used for strapi admin but...
